src/MongoDB.Driver/MongoCredential.cs

```diff
@@ -524,6 +536,30 @@ private static MongoCredential FromComponents(string mechanism, string source, s
                         mechanism,
                         new MongoInternalIdentity(source, username),
                         evidence);
+                case "MONGODB-AWS":
+                    // MUST be "$external". Defaults to $external.
+                    EnsureNullOrExternalSource(mechanism, source);
+                    if (username == null)
+                    {
+                        if (evidence != null && evidence is PasswordEvidence)
+                        {
+                            throw new ArgumentException("A MONGODB-AWS must have access key id.");
+                        }
+
+                        return new MongoCredential(
+                            mechanism,
+                            new MongoExternalAwsIdentity(),
+                            evidence);
+                    }
+                    if (evidence == null || evidence is ExternalEvidence)
+                    {
+                        throw new ArgumentException("A MONGODB-AWS must have secret access key.");
+                    }
```

Comment:
Regarding validation please refer to spec [(link)](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#obtaining-credentials):

> regardless of how Drivers obtain credentials the only valid combination of credentials is an access key ID and a secret access key or an access key ID, a secret access key and a session token.

Reply:
This message about needing secret access key appears often enough that I wonder if we should have a constant that defines it?